pkg/cloud/services/compute/instance.go

```diff
@@ -377,15 +356,29 @@ func (s *Service) createInstanceImpl(eventObject runtime.Object, openStackCluste
 	}
 
 	record.Eventf(eventObject, "SuccessfulCreateServer", "Created server %s with id %s", createdInstance.Name(), createdInstance.ID())
+
+	// iterate over blockDevices and update the metadata
+	opts := volumes.UpdateOpts{
+		Metadata: map[string]string{
+			// add openstack.infrastructure.cluster.x-k8s.io/cluster-uuid
+			"openstack.infrastructure.cluster.x-k8s.io/machine-uuid": createdInstance.ID(),
```

Comment:
this is not the instance ID that we want but the OpenStackMachine ID, a mistake. I'll fix this.